### PR TITLE
add warning message when length exceeds LRECL

### PR DESCRIPTION
**Pull request by tiantn**
Signed-off-by: my-dlq <mynamedlq@163.com>

## Release Notes

Milestone:

Changelog:

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [ ] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [ ] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

## Reminder

After a PR is merged into the `master` branch, create a PR from `master` to the `next` branch resolving any conflicts.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by tiantn** [comment]
fix #1707

File: packages/zowe-explorer-ftp-extension/src/ZoweExplorerFtpMvsApi.ts [hunk 1]
```diff
@@ -141,7 +150,27 @@ export class FtpMvsApi extends AbstractFtpApi implements ZoweExplorerApi.IMvs {
...
             }
-            await DataSetUtils.uploadDataSet(connection, targetDataset, transferOptions);
+            const lrecl: number = dsAtrribute.apiResponse.items[0].lrecl;
+            const data = fs.readFileSync(inputFilePath, "UTF-8");
+            const lines = data.split(/\r?\n/);
+            const foundIndex = lines.findIndex((line) => line.length > lrecl);
+
+            if (foundIndex !== -1) {
+                const select = await vscode.window.showWarningMessage(
+                    "zftp Warning: The " +
```
Comment (std4lqi) [inline packages/zowe-explorer-ftp-extension/src/ZoweExplorerFtpMvsApi.ts:160]: At least one line, like line 10, is longer than dataset LRECL, 80. The exceeding part will be truncated. Do you want to continue?

File: packages/zowe-explorer-ftp-extension/src/ZoweExplorerFtpMvsApi.ts [hunk 2]
```diff
@@ -112,8 +114,15 @@ export class FtpMvsApi extends AbstractFtpApi implements ZoweExplorerApi.IMvs {
         };
         const file = path.basename(inputFilePath).replace(/[^a-z0-9]+/gi, "");
         const member = file.substr(0, 8);
-        let targetDataset;
-        const dsAtrribute = await this.dataSet(dataSetName);
+        let targetDataset: string;
+        const end = dataSetName.indexOf("(");
+        let newDataSetName: string;
```
Comment (std4lqi) [inline packages/zowe-explorer-ftp-extension/src/ZoweExplorerFtpMvsApi.ts:119]: How about renaming `newDataSetName` => `dataSetNameWithoutMember`?

File: packages/zowe-explorer-ftp-extension/src/ZoweExplorerFtpMvsApi.ts (hunk 1 above)
Comment (tiantn) [reply packages/zowe-explorer-ftp-extension/src/ZoweExplorerFtpMvsApi.ts]: Thank you @std4lqi ! I updated the message and the variable name.

**Pull request closed without merging**